compiler/rustc_mir/src/const_eval/eval_queries.rs

```diff
@@ -209,14 +210,28 @@ pub fn eval_to_const_value_raw_provider<'tcx>(
     key: ty::ParamEnvAnd<'tcx, GlobalId<'tcx>>,
 ) -> ::rustc_middle::mir::interpret::EvalToConstValueResult<'tcx> {
     // see comment in const_eval_raw_provider for what we're doing here
-    if key.param_env.reveal() == Reveal::All {
-        let mut key = key;
-        key.param_env = key.param_env.with_user_facing();
-        match tcx.eval_to_const_value_raw(key) {
-            // try again with reveal all as requested
-            Err(ErrorHandled::TooGeneric) => {}
-            // deduplicate calls
-            other => return other,
+    match key.param_env.reveal() {
+        Reveal::Selection => {}
+        Reveal::UserFacing => {
+            let mut key = key;
+            key.param_env = key.param_env.with_reveal_selection();
+            match tcx.eval_to_const_value_raw(key) {
+                // try again with reveal all as requested
```

Comment:
This will try again with reveal "user facing" though, won't it?

Reply:
yeah, didn't update all the comments yet. Opened this case I wanted other people to look at this PR in case there's a fundamental issue I've missed.